helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -679,6 +710,40 @@ public String getClusterName() {
     return _clusterName;
   }
 
+  /**
+   * CloudEventListener
+   */
+  @Override
+  public void onPause(Object eventInfo) {
+    _cloudEventCallbackProperty.onPause(this, eventInfo, _callbackImplClass);
+  }
+
+  @Override
+  public void onResume(Object eventInfo) {
+    _cloudEventCallbackProperty.onResume(this, eventInfo, _callbackImplClass);
+  }
+
+  @Override
+  public ListenerType getListenerType() {
+    return ListenerType.UNORDERED;
+  }
+
+  private AbstractCloudEventCallbackImpl loadCloudEventCallbackImplClass(String implClassName) {
+    AbstractCloudEventCallbackImpl implClass;
+    try {
+      LOG.info("Loading class: " + implClassName);
+      implClass = (AbstractCloudEventCallbackImpl) HelixUtil.loadClass(getClass(), implClassName)
+          .newInstance();
+    } catch (Exception e) {
+      String errMsg = String
+          .format("No cloud event callback implementation class found for: %s. message: ",
+              implClassName);
+      LOG.error(errMsg, e);
+      throw new HelixException(String.format(errMsg, e));
```

Comment:
I am thinking about would this be a critical thing. Let's say if the class is not load, can we log error but let others work? Then we can just use default abstract class as the usage.

Reply:
I guess we can do this. I'll take a note that this is another behavior we should clearly documented in user manual.